samples/apps/copilot-chat-app/webapi/Skills/ChatSkill.cs

```diff
@@ -154,12 +182,63 @@ await foreach (var memory in results)
 
         // Update the token limit.
         memoryText = $"Past memories (format: [memory type] <label>: <details>):\n{memoryText.Trim()}";
-        tokenLimit -= Utils.TokenCount(memoryText);
+        tokenLimit -= Utilities.TokenCount(memoryText);
         context.Variables.Set("tokenLimit", tokenLimit.ToString(new NumberFormatInfo()));
 
         return memoryText;
     }
 
+    /// <summary>
+    /// Extract relevant additional knowledge using a Semantic Kernel planner.
+    /// </summary>
+    [SKFunction("Acquire external information")]
+    [SKFunctionName("AcquireExternalInformation")]
+    [SKFunctionContextParameter(Name = "userIntent", Description = "The intent of the user.")]
+    [SKFunctionContextParameter(Name = "tokenLimit", Description = "Maximum number of tokens")]
+    public async Task<string> AcquireExternalInformationAsync(SKContext context)
+    {
+        if (!this._plannerOptions.Enabled)
+        {
+            return string.Empty;
+        }
+
+        SequentialPlanner planner = await this._plannerFactory(this._kernel);
+
+        // Create a plan and run it.
+        Plan plan = await planner.CreatePlanAsync(context["userIntent"]);
+        while (plan.HasNextStep)
+        {
+            // WORKAROUND for an issue when the planner returns a plan containing a step that is a plan with no steps.
```

Comment:
Phrase this as a TODO?
TODO: Reset the plan result more gracefully when the planner returns a plan with no steps.
Running these invalid steps resets the plan result (plan.State.Input) to the original input/goal statement.

Reply:
This is a workaround for a bug in SequentialPlanner. I'll add the term TODO to remove the workaround. :)